students/ChrisH/lesson05/mailroom.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+# -----------------------------------------------------------
+# mailroom.py
+#  Part 3. Script to automate writing thank you emails to donors.
+#  Add exceptions and use comprehensions.
+# -----------------------------------------------------------
+
+import time
+
+from collections import namedtuple
+
+# Giving namedtuples a shot
+Donor = namedtuple('Donor', 'first,last')
+
+
+# Global data structure
+donor_data = \
+    {Donor('Al', 'Donor1'): [10.00, 20.00, 30.00, 40.00, 50.00],
+     Donor('Bert', 'Donor2'): [10.00],
+     Donor('Connie', 'Donor3'): [10.00, 10.00, 10.01],
+     Donor('Dennis', 'Donor4'): [10.00, 20.00, 20.00],
+     Donor('Egbert', 'Donor5'): [10.39, 20.21, 10.59, 4000.40],
+     }
+
+
+def get_donor_fullname(donor):
+    """
+    Given a donor namedtuple, assembles and returns the donor's full name.
+    :param donor: namedtuple type: Donor
+    :return: string with donor's full name
+    """
+    return (donor.first + ' ' + donor.last).strip()   # Strip, in case first name is blank
+
+
+def menu(menu_data):
+    """
+    Prints the main user menu & retrieves user selection.
+    :param: a menu dictionary, key as numbered list.
+            Dict key 0 should be a null function (or text for bad choice)
+            Dict value is a list with the following entries:
+                [0] = Text for user prompt
+                [1] = Function that should be run for that choice
+    :return: the function corresponding to the user's selection
+    """
+    print("\nPlease choose one of the following options:")
+    for n in range(1, len(menu_data)):                     # Start at item 1, item 0 should be null function
```

Comment:
Change this to `for n in range(1, len(menu_data) + 1):  ` and you don't need the `0` menu option